### PR TITLE
Module Unification Addon blueprint

### PR DESCRIPTION
**Pull request by cibernox**
This adds the ability of generating addons using the module unification layout.

MODULE_UNIFICATION=true ember addon my-addon
MODULE_UNIFICATION=true ember addon my-addon --yarn

@GavinJoyce @mixonic

### CODE REVIEW

File: blueprints/module-unification-addon/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    let name = stringUtil.dasherize(rawName);
+    let namespace = stringUtil.classify(rawName);
+    let addonName = stringUtil.dasherize(rawName);
+
+    return {
+      name,
+      modulePrefix: name,
+      namespace,
+      addonName,
+      // addonModulePrefix: addonName,
```
Comment (cibernox) [inline blueprints/module-unification-addon/index.js]: It's not clear to me the difference between `addonModulePrefix` and `addonName` and which one should I use.

**Comment by cibernox** [comment]
Can someone rebuild this? I think it's a random failure.

File: blueprints/module-unification-addon/index.js (hunk 1 above)
Comment (mixonic) [reply blueprints/module-unification-addon/index.js]: I don't see `addonModulePrefix` used anywhere in these files.

Comment (cibernox) [reply blueprints/module-unification-addon/index.js]: It is not, I decided to use addon name. But I wanted to understand if there was a good reason for that weird naming

File: tests/fixtures/module-unification-addon/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
+
+Copyright (c) 2018
```
Comment (ro0gr) [inline tests/fixtures/module-unification-addon/LICENSE.md]: it will fail each year :)

File: tests/fixtures/module-unification-addon/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "test": "ember test",
+    "test:all": "ember try:each"
+  },
+  "dependencies": {
+    "ember-cli-babel": "^6.6.0"
+  },
+  "devDependencies": {
+    "broccoli-asset-rev": "^2.4.5",
+    "ember-ajax": "^3.0.0",
+    "ember-cli": "~3.1.0-beta.1",
```
Comment (ro0gr) [inline tests/fixtures/module-unification-addon/package.json]: it will fail each release

Comment (cibernox) [reply tests/fixtures/module-unification-addon/package.json]: The test is testing that the same files are being created, not that the contents of those files is the same. Afaik, all other tests are similar.

Comment (ro0gr) [reply tests/fixtures/module-unification-addon/package.json]: oh, right. there is no verification of result files contents

File: blueprints/module-unification-addon/index.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+const addonBlueprint = require('../addon');
+
+module.exports = Object.assign({}, addonBlueprint, {
```
Comment (cibernox) [inline blueprints/module-unification-addon/index.js]: This is where it all happens. The only difference with the generate of standard addons is the description and the files. I couldn't find a better way of "inheriting" from an existing blueprint.

File: blueprints/module-unification-addon/files/.travis.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+---
```
Comment (rwjblue) [inline blueprints/module-unification-addon/files/.travis.yml]: Seems weird to have _both_ `.travis.yml` and `travis.yml` in the diff here

Comment (rwjblue) [reply blueprints/module-unification-addon/files/.travis.yml]: seems like all of the `.foo` files have this issue?

**Comment by rwjblue** [comment]
Chatted with @mixonic in slack, we agree that this can land and changing the default dummy app setup to be module unification should happen in a follow up PR.

**Pull request merged**